### ISSUE TITLE
Should be abandoned due to PSR?

### ISSUE THREAD

**Issue by someniatko** [brick/http#4]
Nowadays we have 
- [PSR-7 HTTP Message Interface](https://www.php-fig.org/psr/psr-7)
- [PSR-15 HTTP Handlers](https://www.php-fig.org/psr/psr-15)

which cover the functionality of this lib. Therefore IMO it is wise to abandon the lib in favor of PSRs, and the existing users should be encouraged to use it instead. So we don't have yet another HTTP request abstraction, and also this will reduce the maintenance burden for you.

**Comment by BenMorel** [comment]
Hi,

> which cover the functionality of this lib

This library is an implementation of HTTP messages, the PSRs you link to are interfaces for such messages. So they're not actually redundant. The real questions are:

- should this library implement these PSRs;
- should this library be abandoned in favour of symfony/http-foundation or others.

> the existing users should be encouraged to use it instead

I don't think there are any users of this library at the moment. The README is empty, and most if not all downloads are from my own projects.

> Therefore IMO it is wise to abandon the lib

While I do agree that this library is mostly redundant with other popular libraries, I'm still using it in a project that I'm not ready to migrate to Symfony just yet. In the meantime, is there anything I can do to make it more obvious that this library is not something you should choose for a new projet? (I think it is quite obvious already, as I said there probably aren't any users of this library besides me).

**Comment by someniatko** [comment]
Yeah, this lib provides both the abstraction and the implementation, and PSRs cover only abstractions indeed, I should have mentioned it in the initial issue post.

> In the meantime, is there anything I can do to make it more obvious that this library is not something you should choose for a new project?

I think marking it as "abandoned" in Packagist, and mentioning it in the README should work fine.

**Comment by someniatko** [comment]
> as I said there probably aren't any users of this library besides me

Well I am an active user of your `date-time` and `money` libs, and that made me explore other libraries in the `brick/` namespace, that's why I stumbled upon this lib. Nothing here suggests it is actually more like a personal project :)

**Comment by BenMorel** [comment]
> I think marking it as "abandoned" in Packagist, and mentioning it in the README should work fine.

I don't think I can release new versions if the package is abandoned, so this will probably not work. I can update the README though.

**Comment by BenMorel** [comment]
> Nothing here suggests it is actually more like a personal project :)

Yeah, it was not supposed to be a personal project, no more than `brick/date-time` or `brick/money`. It just turned out to be much less useful!
